lib/matplotlib/dviread.py

```diff
@@ -58,10 +58,70 @@
 
 # The marks on a page consist of text and boxes. A page also has dimensions.
 Page = namedtuple('Page', 'text boxes height width descent')
-Text = namedtuple('Text', 'x y font glyph width')
 Box = namedtuple('Box', 'x y height width')
 
 
+# Also a namedtuple, for backcompat.
+class Text(namedtuple('Text', 'x y font glyph width')):
+    """
+    A glyph in the dvi file.
+
+    The *x* and *y* attributes directly position the glyph.  The *font*,
+    *glyph*, and *width* attributes are kept public for back-compatibility, but
+    users wanting to draw the glyph themselves are encouraged to instead load
+    the font specified by `font_path` at `font_size`, warp it with the effects
+    specified by `font_effects`, and load the glyph specified by `glyph_spec`.
+    """
+
+    def _get_pdftexmap_entry(self):
+        return PsfontsMap(find_tex_file("pdftex.map"))[self.font.texname]
+
+    @property
+    def font_path(self):
+        """The `~pathlib.Path` to the font for this glyph."""
+        psfont = self._get_pdftexmap_entry()
+        if psfont.filename is None:
+            raise ValueError("No usable font file found for {} ({}); "
+                             "the font may lack a Type-1 version"
+                             .format(psfont.psname.decode("ascii"),
+                                     psfont.texname.decode("ascii")))
+        return Path(psfont.filename)
+
+    @property
+    def font_size(self):
+        """The font size."""
+        return self.font.size
+
+    @property
+    def font_effects(self):
+        """
+        The "font effects" dict for this glyph.
+
+        This dict contains the values for this glyph of SlantFont and
+        ExtendFont (if any), read off :file:`pdftex.map`.
+        """
+        return self._get_pdftexmap_entry().effects
+
+    @property
+    def glyph_spec(self):
+        """
+        Either the glyph name or the native charmap glyph index.
```

Comment:
I'm not quite convinced that this type instability str/int is a reasonable API.

The usage does awkward type checks

```
if isinstance(spec, str):
    ...
elif isinstance(spec, int):
    ...
else:  # Should not occur.
   raise TypeError(f"Glyph spec of unexpected type: {spec!r}")
```
Would it be better to have `glyph_name` and `glyph_index` properties and do something like
```
if text.glyph_name is not None
    ...
elif text.glyph_index is not None:
    ...
else:  # Should not occur.
   raise ...
```

Reply:
I think that `glyph_spec` is of type `str | int` is simpler to document than "`glyph_name` is of type `str | None` *and* `glyph_index` is of type int | None *and* exactly one of `glyph_name` and `glyph_index` is not None".  (e.g., in a statically typed language, you could usually represent the first constraint but not the second one.)

Reply:
> and exactly one of glyph_name and glyph_index is not None

Well you can write it down in the docs :). I don't think it's important to enforce this structurally. I'm more concerned with the opaque type acrobatics this forces on the usage side.

Reply:
Let's discuss this on a call (... once I have time again to attend one :/ perhaps next week)

Reply:
See my comment below on `glyph_name_or_index`. Maybe that's good enough. It's not as if many people will use this API and we can always add `glyph_name` and `glyph_index` properties later.